frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java

```diff
@@ -932,13 +946,35 @@ private static int getAvailableGpu() {
                 for (int i = 1; i < list.size(); i++) {
                     gpuIds.add(Integer.parseInt(list.get(i)));
                 }
-            }
+                }catch (IOException | InterruptedException e) {
+                System.out.println("nvidia-smi not available or failed: " + e.getMessage());
+                }
+                try {
+                Process process = Runtime.getRuntime().exec("xpu-smi discovery --dump 1");
+                int ret = process.waitFor();
+                    if (ret != 0) {
+                        return 0;
+                    }
+                List<String> list =
+                        IOUtils.readLines(process.getInputStream(), StandardCharsets.UTF_8);
+                if (list.isEmpty() || !list.get(0).contains("Device ID")) {
+                    throw new AssertionError("Unexpected xpu-smi response.");
+                }
+                for (int i = 1; i < list.size(); i++) {
+                    gpuIds.add(Integer.parseInt(list.get(i)));
+                }
+                }catch (IOException | InterruptedException e) {
+                System.out.println("xpu-smi not available or failed: " + e.getMessage());
```

Comment:
Is Intel GPU also set [CUDA_VISIBLE_DEVICES](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java#L891-L892)? If not, this part code will not be executed.

Reply:
They are not set with CUDA_VISIBLE_DEVICES. We have updated this part. Thank you for the suggestion!

Reply:
They will not be set, and we've removed this part. Thanks!